concepts/observability-strategy/configurable-tracing/pocs/pipeline-investigations/README.md

```diff
@@ -0,0 +1,469 @@
+  - 1GiB Memory
+  - 1 CPU
+  - Memory hard limit configured 75% of total memory, memory spike limit configured to the 10% of total available memory.
+  - Retry backoff limit 30 seconds and maximum retry time to 300 seconds.
+  - Bath size 512 spans per batch, maximum batch queue size 512 batches.
+
+## Observations
+
+### Resource Usage
+
+Purpose of this test is observing collector behaviour under high load, resource consumption, and out of memory situation. Test was running over 12 hours with constant load generation.
+
+| ![CPU Utilization](assets/cpu_usage_long.png) |
+| :--: |
+| Fig. 1 CPU utilization |
+
+| ![Memory Utilization](assets/memory_usage_long.png) |
+| :--: |
+| Fig. 2 Memory utilization |
+
+- As `Fig. 1` shown, CPU utilization stay moderate during whole test execution time.
+- `Fig. 2` shown memory consumption over 12 hours, the collector never run OOM situation.
+
+| ![Refused Spans](assets/refused_spans_long.png) |
+| :--: |
+| Fig. 3 Refused spans |
+
+- `Fig. 3` show the behaviour of memory limiter in case of high load and high memory utilization. Memory limiter kick in when the memory utilization get above defined `hard limit` and start refuse incoming data to avoid OOM situation.
+
+### Retry and Queuing Test
+
+This scenario tests an outage of the backend. In case of a backend outage data should be queued for max `300 seconds` and retried to export before data is dropped from queue.
+
+| ![Batch Queue](assets/queue_test_long.png) |
+| :--: |
+| Fig. 4 Batch queue |
+
+The `Fig. 4` shows the queue size is going up in time and going down approximately every 5 minutes. This is because queued data will be dropped when could not be exported successfully after 300 second (5 Minutes).
+
+| ![Failed Exports](assets/failed_send_spans_long.png) |
+| :--: |
+| Fig. 5 Failed exports |
+
+The `Fig. 5` show unsuccessful export attempts during test execution.
+
+### Test with CLS Backend
+
+The throughput test runs over one hour to measure average throughput of span ingestion. In the peak times, no span should be dropped or refused by the pipeline and overall resource consumption should stay moderate.
+The test uses a CLS instance without any configuration changes as backend. The test load generator is started with 2 concurrent producers, which produce on average `1550 spans/sec`.
+
+| ![Accepted Spans](assets/accepted_spans_cls.png) |
+| :--: |
+| Fig. 6 Accepted spans |
+
+| ![Send Exports](assets/send_spans_cls.png) |
+| :--: |
+| Fig. 7 Send exports |
+
+| ![Queue Size](assets/queue_size_cls.png) |
+| :--: |
+| Fig. 8 Queue size |
+
+| ![CPU Utilization](assets/cpu_cls.png) |
+| :--: |
+| Fig. 9 CPU Utilization |
+
+| ![Memory Utilization](assets/memory_cls.png) |
+| :--: |
+| Fig. 10 Memory Utilization |
+
+The CLS instance became approximately after 4 minutes unavailable, which can be observed from `Fig. 8`. After 4 minutes of load test, the queue size starts to grow because spans can't be successfully exported.
```

Comment:
```suggestion
The CLS instance became unavailable after approximately 4 minutes, which can be observed from `Fig. 8`. After 4 minutes of load test, the queue size starts to grow because spans can't be successfully exported.
```